packages/babel-parser/src/plugins/typescript.js

```diff
@@ -794,6 +795,21 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         return true;
       }
 
+      if (this.match(tt.bracketL)) {
```

Comment:
The two checks should be merged, since there could be code like this:
```js
type Foo = ([{a}]: any) => {};
```

Reply:
Currently it just works for `[a]` `{a}` and `[{a}]`. Is that neccesary to check brace and bracket within a same if block? Since there is only brace and bracket, no more conditions.